### PR TITLE
Dashboard and company items presentation

### LINKED ISSUE #24: Allow representative to display items
A list of all items a company has should be view able but not editable, basic sorting should exist

### LINKED ISSUE #49: Show companies as part of dashboard
When the user logs in the first page they should see should be the dashboard.
On the dashboard the companies that are represented should be displayed so a selection can be made on what items to show.
Child of #24

### LINKED ISSUE #51: Indicate that dashboard is loading
Have some visual representation that dashboard is loading.
Preferably do this by creating some dummy company cards and have a spinner image in them so that it is obvious, something similar to what facebook does.
Child of #49

### LINKED ISSUE #53: Allow for item display
When clicking on an item on the list it should display further details of the items

### LINKED ISSUE #54: Create tests for dashboard

### LINKED ISSUE #55: Navigation: From Company items to dashboard breaks dashboard loading
Steps to reproduce.

1. Navigate to a company
2. Navigate back to home
3. No loading appears but after 2 seconds the screen reloads

### PR DESCRIPTION
**Pull request by HarryPi**
## Features added
- Dashboard created
- Show companies on dashboard
- Show companies items
- Allow for items to be sorted
- Implemented Search functionality for items based on description and name. ( Can perhaps consider more filters later on #56)

## Known bugs
- ~~Navigation from Company to dashboard not loading. See #55~~ Fixed on 94cb3394ecb547f14679b9d106ca618722a0aa52

Closes #24, closes #49, Closes #51, Closes #53, Closes #54, Fixes #55

### CODE REVIEW

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <component name="InspectionProjectProfileManager">
   <profile version="1.0">
     <option name="myName" value="Project Default" />
+    <inspection_tool class="Eslint" enabled="true" level="WARNING" enabled_by_default="true" />
```
Comment (markscamilleri) [inline .idea/inspectionProfiles/Project_Default.xml:4]: We don't need to commit anything in the .idea folder as this is intellij metadata. I think it's best we remove these from git and add to gitignore reducing the size of PRs

File: RepClient/src/app/company/company.page.spec.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    fixture = TestBed.createComponent(CompanyPage);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  }));
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
```
Comment (markscamilleri) [inline RepClient/src/app/company/company.page.spec.ts:28]: Any reason you're not testing sortBy, itemLookup and cancelLookup?

**Comment by markscamilleri** [comment]
Also I think you might be using an old version of the Travis file. (And there's merge conflicts so you need to pull master anyway)

File: RepClient/src/app/home/home.page.scss [hunk 3]
```diff
@@ -1,4 +1,4 @@
-.welcome-card img {
-  max-height: 35vh;
-  overflow: hidden;
+ion-spinner {
+  width: 100%;
+  transform: scale(3);
```
Comment (DominikWas) [inline RepClient/src/app/home/home.page.scss:3]: Seen you use this styling previously (e.g. company.page.scss). Consider using a universal style sheet

File: RepClient/src/main.ts [hunk 4]
```diff
@@ -1,6 +1,5 @@
 import { enableProdMode } from '@angular/core';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
-import 'hammerjs';
```
Comment (DominikWas) [inline RepClient/src/main.ts:3]: Is this touch gesture package redundant now?

File: .idea/inspectionProfiles/Project_Default.xml (hunk 1 above)
Comment (HarryPi) [reply .idea/inspectionProfiles/Project_Default.xml:4]: Yh that seems to be a mistake on my gitignore file on how i spelled the ignore idea one

File: RepClient/src/app/company/company.page.spec.ts (hunk 2 above)
Comment (HarryPi) [reply RepClient/src/app/company/company.page.spec.ts:28]: More functionality to be added later this was more of an optional thing. Currently the test code would be a duplication of the .ts code as it checks and displays dummy data

File: RepClient/src/app/home/home.page.scss (hunk 3 above)
Comment (HarryPi) [reply RepClient/src/app/home/home.page.scss:3]: No because i do not want all spinners in all page to appear like this. This is only for specific full length wide items. There is a further plan to make a component for the loading as that is the same code as well thus transferring the style to only one place.

File: RepClient/src/main.ts (hunk 4 above)
Comment (HarryPi) [reply RepClient/src/main.ts:3]: No, this is how you used to use hammerjs prior to Angular 9 after angular 9 you instead use Hammerjs Module and import it that way. Hammerjs is still the default gesture control for angular see https://github.com/angular/angular/issues/35547#issuecomment-588277881 from angular team

File: RepClient/src/app/company/company.page.spec.ts (hunk 2 above)
Comment (HarryPi) [reply RepClient/src/app/company/company.page.spec.ts:28]: > I see you've gone from a genetic list page which you can use for multiple reasons to a specific company page

It really depends on what you need it for. As an example one list is virtual and the other is a normal list. This on its own brings the issue that you will have 3 templates in one component that you will need to maintain. A virtual list is a list that un-loads any item to keep the memory usage on a mobile phone efficient. On the other hand a company wouldnt need a virutal list as it would be an over kill. Then on one you would need to hook it up with the search etc, that would need event binding / exposure.
TLDR; you should use generic components yes, but in this case it would require 10x times the work as what i was after was very simple. What should be a generic component and its planned to be is the loading part :)

File: .idea/inspectionProfiles/Project_Default.xml (hunk 1 above)
Comment (markscamilleri) [reply .idea/inspectionProfiles/Project_Default.xml:4]: `.idea/` is good, but you need to remove the file from git, usually with `git rm` if I remember correctly (it won't physically delete the file)

Comment (HarryPi) [reply .idea/inspectionProfiles/Project_Default.xml]: Ah my bad forgot to do that 

> usually with `git rm`

I think in cases like this you need to use the --cache like this `git rm --cached -r .idea` . Anyhow i think it should be fine now.

File: .travis.yml [hunk 5]
```diff
@@ -3,7 +3,7 @@ sudo: true
 
 language: node_js
 node_js:
-  - "10.9"
+  - "10.13"
```
Comment (markscamilleri) [inline .travis.yml:6]: Might you be using an old version? Please pull the master branch into this branch first

File: RepClient/src/app/app-routing.module.ts [hunk 6]
```diff
@@ -1,33 +1,33 @@
...
+  {
+    canActivate: [AuthGuard],
+    loadChildren: () => import('./home/home.module').then(m => m.HomePageModule),
+    path: 'home'
+  },
+  {
+    loadChildren: () => import('./auth/auth.module').then(m => m.AuthPageModule),
+    path: 'auth'
+  },
  {
```
[the remark this answers is not in the document]
Comment (HarryPi) [reply RepClient/src/app/app-routing.module.ts:19]: That is probably a BOM character as it doesnt exist in my code and all my changes are pushed

File: .travis.yml (hunk 5 above)
Comment (HarryPi) [reply .travis.yml]: Yes that was my bad.
